### PR TITLE
KAFKA-2390-followup; add unit test for OffsetOutOfRange exception

### PR DESCRIPTION
**Pull request by lindong28**

### CODE REVIEW

**Comment by hachikuji** [comment]
@lindong28 Why not include this test in FetcherTest? Also, do you think it's worthwhile adding a test case which checks that the out of range error is ignored when the user seeks to a different offset?

**Comment by lindong28** [comment]
@hachikuji I couldn't include this test in FetcherTest, because to test OffsetOutOfRange exception the defaultResetPolicy needs to be NONE. But the defaultResetPolicy can't be changed after setup.

Sure, let me add another test to check that the out of range error is ignored when the user seeks to a different offset.

**Comment by hachikuji** [comment]
@lindong28 Not sure that problem calls for a new class. An easy way would be to create a factory method in FetcherTest which each test uses to construct a new Fetcher instance. Up to you, but it's nice to keep related tests organized together.

**Comment by lindong28** [comment]
@hachikuji I agree that nice to keep related tests organized together. I can implement that factory class to be used across test cases.

Another alternative is to add method setOffsetResetStrategy() in class SubscriptionState. That looks OK to me and would make things much easier here. What do you think?

**Comment by lindong28** [comment]
@hachikuji I have moved the test cases to FetcherTest. Please let me know if it looks good. Thank you.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java [hunk 1]
```diff
@@ -73,27 +75,15 @@
...
-                                                                          new ByteArrayDeserializer(),
-                                                                          metadata,
-                                                                          subscriptions,
-                                                                          metrics,
-                                                                          "consumer" + groupId,
-                                                                          metricTags,
-                                                                          time,
-                                                                          retryBackoffMs);
+    private Fetcher<byte[], byte[]> fetcher = createFetcher(subscriptions, metrics);
+    private Fetcher<byte[], byte[]> fetcherNoAutoReset = createFetcher(subscriptionsNoAutoReset, new Metrics(time));
```
Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java:86]: Might be no need to make this a field since it's only used in two test cases.

**Comment by hachikuji** [comment]
@lindong28 One minor comment. Overall, LGTM.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java (hunk 1 above)
Comment (guozhangwang) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java:86]: Sounds OK to me since it is used more than once.

**Comment by lindong28** [comment]
Thanks @hachikuji @guozhangwang for your quick review.
